source/services/helper.js

```diff
@@ -169,7 +169,7 @@
 
                 // cloneEnabled and cross-tree so copy and do not remove from source
                 if (this.isClone() && this.isForeign()) {
-                  this.parent.insertNode(this.index, angular.copy(nodeData));
+                  this.parent.insertNode(this.index, this.sourceInfo.cloneModel);
```

Comment:
In `cloneModel()`, we again check if cloning is enabled, which means we perform the check 2 times (in `helper.js` on L127 and here on L172).

Do you have any suggestions on how we can refactor this to prevent double checking?